### ISSUE TITLE
[clang-tidy] `readability-else-after-return` not detected in `case` blocks

### ISSUE THREAD

**Issue by firewave** [llvm/llvm-project#160033]
```cpp
void f(int i, bool b)
{
    switch(i)
    {
        case 0:
            if (b) {
                return;
            }
            else {}
        break;
    }
}
```
https://godbolt.org/z/fG1z7b97d

**Comment by flovent** [comment]
This is probably because the parent node of the If statement is not `CompoundStmt`. The code below will report a warning.
```
void f(int i, bool b)
{
    switch(i)
    {
        case 0: {
            if (b) {
                return;
            }
            else {}
        }
        break;
    }
}
```
https://godbolt.org/z/Tcrrec9zY
